### DISCUSSION TITLE
Is it a bug ?

### DISCUSSION THREAD

**Discussion by NguyenBentley** [iTwin/itwinjs-core#129] [General] [answered]
I do the test

1. The document of the methode "ColorDef.fromString" 
...
    const aBlue = ColorDef.fromString("#0000FF"); // must be ColorDef.blue
    assert.equal(aBlue, ColorDef.blue);
    const aRed = ColorDef.fromString("#FF0000"); // must be ColorDef.red
    assert.equal(aRed, ColorDef.red);
    const aBlack = ColorDef.fromString("#000000"); // must be ColorDef.black
    assert.equal(aBlack, ColorDef.black);
    const aWhite = ColorDef.fromString("#FFFFFF"); // must be ColorDef.white
    assert.equal(aWhite, ColorDef.white);
...
all the test are OK

The document of the methode "ColorDef.fromString" says
/** Create a ColorDef from a string representation. The following representations are supported:
   * *"rgb(255,0,0)"*
   * *"rgba(255,0,0,.2)"*
   * *"rgb(100%,0%,0%)"*
   * *"hsl(120,50%,50%)"*
   * *"#rrbbgg"*
   * *"blanchedAlmond"* (see possible values from [[ColorByName]]). Case-insensitive.
   */
  public static fromString(val: string): ColorDef {
    return this.fromTbgr(this.computeTbgrFromString(val));
  }

I think the document of the methode should be   * *"rrggbb"*

2. Green is not green
then I do

...
   const aGreen = ColorDef.fromString("#00FF00"); // must be ColorDef.green
    assert.equal(aGreen, ColorDef.green);
...

the test Failed !
Why green is not green ?

**Comment by pmconne** [comment]
`ColorDef.green` is produced from `ColorByName.green` which for historical reasons is `#008000`, not `#00ff00`. That has surprised me before as well. It's primarily used for producing decoration graphics, for which a medium green is generally more appropriate than super-bright green.

**Reply by pmconne** [reply]
You are correct about the documentation - `#rrbbgg` should be `#rrggbb`.

**Comment by NguyenBentley** [comment, marked as answer]
Thank
Since my firm intend to use iModelJs for the next product and since we are new to iModelJs so i must do a lot of offline & integration test to check that we get that what we just write to iModelJs
Now i know i must take into account some historical features

**Reply by kabentley** [reply]
Green is #008000 for [compatibility with HTML](https://www.w3schools.com/colors/colors_hex.asp)

**Reply by NguyenBentley** [reply]
OK
